compiler/fir/resolve/src/org/jetbrains/kotlin/fir/declarations/FirAnnotationUtils.kt

```diff
@@ -88,3 +93,26 @@ private val DEFAULT_USE_SITE_TARGETS: Set<AnnotationUseSiteTarget> =
 fun FirAnnotatedDeclaration.hasAnnotation(classId: ClassId): Boolean {
     return annotations.any { it.toAnnotationClassId() == classId }
 }
+
+private val DEPRECATED_ANNOTATION = ClassId(FqName("kotlin"), Name.identifier("Deprecated"))
+
+fun FirAnnotatedDeclaration.deprecationStatus(): DeprecationLevel? {
+    for (annotation in annotations) {
+        // TODO: also check Kotlin version annotations
+        if (annotation.toAnnotationClassId() != DEPRECATED_ANNOTATION) continue
+
+        val deprecatedClass = annotation.getCorrespondingClassSymbolOrNull(session)!!
+        val deprecatedConstructor = deprecatedClass.fir.declarations.single { it is FirConstructor } as FirConstructor
+        val (_ /*message*/, _ /*replaceWith*/, level) = annotation.argumentMapping?.let { mapping ->
```

Comment:
That's not actually correct because annotation arguments are resolved in the same transformer as function calls.